### PR TITLE
feat: include falsy json fields

### PR DESCRIPTION
**Pull request by toddbaert**
- adds custom JSON config which returns all falsy json values in responses
- adds associated tests
- removes now irrelevant doc

Responses that used to look like these:

`{"reason":"DEFAULT", "variant":"red", "metadata":{}}`
`{"reason":"STATIC", "variant":"off", "metadata":{}}`

Now look like these:

`{"value":0, "reason":"DEFAULT", "variant":"red", "metadata":{}}`
`{"value":false, "reason":"STATIC", "variant":"off", "metadata":{}}`

Fixes: https://github.com/open-feature/flagd/issues/787

### CODE REVIEW

File: core/pkg/service/flag-evaluation/connect_service.go [hunk 1]
```diff
@@ -119,7 +120,14 @@ func (s *ConnectService) setupServer(svcConf service.Configuration) (net.Listene
 		s.eventingConfiguration,
 		s.metrics,
 	)
-	path, handler := schemaConnectV1.NewServiceHandler(fes, svcConf.Options...)
+
+	marshalOpts := WithJSON(
+		// json parsing configuration - we emit "unpopulated" fields (falsy fields are not dropped)
+		protojson.MarshalOptions{EmitUnpopulated: true},
+		protojson.UnmarshalOptions{DiscardUnknown: true},
```
Comment (beeme1mr) [inline core/pkg/service/flag-evaluation/connect_service.go:127]: Does this change the unmarshaling behavior from what we had?

**Review by beeme1mr — APPROVED** [review]
Awesome, great job.

File: core/pkg/service/flag-evaluation/connect_service.go (hunk 1 above)
Comment (toddbaert) [reply core/pkg/service/flag-evaluation/connect_service.go:127]: Only in the way we want. `DiscardUnknown` was the previous behavior, see [here](https://github.com/bufbuild/connect-go/blob/main/codec.go#L176).

File: core/pkg/service/flag-evaluation/json_codec.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+		connect.WithCodec(&jsonCodec{name: "json; charset=utf-8", marshal: marshal, unmarshal: unmarshal}),
+	)
+}
+
+type jsonCodec struct {
+	name      string
+	marshal   protojson.MarshalOptions
+	unmarshal protojson.UnmarshalOptions
+}
+
```
Comment (Kavindu-Dodan) [inline core/pkg/service/flag-evaluation/json_codec.go:32]:
```suggestion
var _ connect.Codec = (*jsonCodec)(nil)
```

This is a compile time check to see if `jsonCodec` implements `connect.Codec` interface

**Review by Kavindu-Dodan — APPROVED** [review]
Nice work 🎉

File: core/pkg/service/flag-evaluation/json_codec.go (hunk 2 above)
Comment (Kavindu-Dodan) [reply core/pkg/service/flag-evaluation/json_codec.go:32]: Connect internals also do the same - https://github.com/bufbuild/connect-go/blob/main/codec.go#L146C1-L146C37

File: core/pkg/service/flag-evaluation/json_codec_test.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			jsonMap := map[string]interface{}{}
+			message, err := tt.message()
+			if err != nil {
+				t.Errorf("Got error creating message: %v", err.Error())
+			}
+			bytes, err := tt.marshaller(message)
+			if tt.wantErr && err == nil {
+				t.Errorf("Expected error but not none")
```
Comment (bacherfl) [inline core/pkg/service/flag-evaluation/json_codec_test.go:104]: optional suggestion: To keep consistent with the other assertions, you could also use `require.NotNil(t, err)`, or `require.NotNilf(t, err, "Expected error but not none")`

**Review by bacherfl — APPROVED** [review]
left one optional suggestion, but LGTM otherwise!

File: core/pkg/service/flag-evaluation/json_codec_test.go (hunk 3 above)
Comment (toddbaert) [reply core/pkg/service/flag-evaluation/json_codec_test.go:104]: ty!

Comment (toddbaert) [reply core/pkg/service/flag-evaluation/json_codec_test.go:104]: (I meant to type "got none" also)

**Pull request merged**